kedro-datasets/kedro_datasets/ibis/table_dataset.py

```diff
@@ -115,31 +107,13 @@ def __init__(  # noqa: PLR0913
             metadata: Any arbitrary metadata. This is ignored by Kedro,
                 but may be consumed by users or external plugins.
         """
-        if filepath is None and table_name is None:
-            raise DatasetError(
-                "Must provide at least one of `filepath` or `table_name`."
-            )
-
-        if filepath is not None or file_format is not None:
-            warnings.warn(
-                "Use 'FileDataset' to load and save files with an Ibis "
-                "backend; the functionality will be removed from 'Table"
-                "Dataset' in Kedro-Datasets 6.0.0",
-                KedroDeprecationWarning,
-                stacklevel=2,
-            )
-
-        self._filepath = filepath
-        self._file_format = file_format
+        if table_name is None:
+            raise DatasetError("Must provide `table_name`.")
+
```

Comment:
```suggestion
        if table_name is None:
            raise DatasetError("Must provide `table_name`.")

```
Not necessary; can just make the argument a required, positional argument.